source/parse.h

```diff
@@ -3318,6 +3322,24 @@ class parser
                 }
             }
 
+            if (deduced_type) {
+                auto& type = std::get<declaration_node::object>(n->type);
+                if (peek(1)->type() == lexeme::Ampersand) {
+                    type->address_of = &curr();
+                }
+                else if (peek(1)->type() == lexeme::Multiply) {
+                    type->dereference_of = &curr();
+                    for (int i = 1; peek(i)->type() == lexeme::Multiply; ++i)
+                        type->dereference_cnt += 1;
+                }
+                else if (
+                    (peek(1)->type() == lexeme::LeftParen && curr().type() != lexeme::Colon)
+                    || curr().type() == lexeme::Identifier 
+                ) {
+                    type->suspicious_initialization = &curr();
```

Comment:
It's not clear to me what is suspicious exactly... this probably needs a comment to define what is suspicious. Or maybe a better name?

Reply:
I have added comments to the code.

`suspicious_initialization` is initialization by a function call or variable that must be checked. Variable or function might be defined on the cpp1 side, which is why I have called it `suspicious`. On the other hand, it means that the type needs to be deduced, so maybe there is a better name.